### ISSUE TITLE
[ISSUE] Unable to connect identity to self deployed keycloak

### ISSUE THREAD

**Issue by knottnhupfer** [camunda/camunda-platform-helm#623]
I'm trying to configure my already existing keycloak using the parameters available in the helm chart. My keycloak configuration looks like following:

```
global:
  image:
    tag: 8.2.0
  identity:
    keycloak:
      url:
        protocol: "http"
        host: "keycloak"
        port: 80
      auth:
        adminUser: "admin"
        existingSecret: camunda-platform-secrets
        existingSecretKey: "keycloak-secret"

identity:
  enabled: true
  keycloak:
    legacy: false
    enabled: false

```

Perviously I had a wrong hostname and after fixing that now I get following log output from identity pod.

```

  .   ____          _            __ _ _                                                                                                                                                                                                                                                                                        
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \                                                                                                                                                                                                                                                                                       
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \                                                                                                                                                                                                                                                                                      
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )                                                                                                                                                                                                                                                                                     
  '  |____| .__|_| |_|_| |_\__, | / / / /                                                                                                                                                                                                                                                                                      
 =========|_|==============|___/=/_/_/_/                                                                                                                                                                                                                                                                                       
 :: Spring Boot ::                (v3.0.5)                                                                                                                                                                                                                                                                                     

2023-04-21 20:22:10.518  INFO 1 --- [           main] i.c.i.Application                        : Starting Application using Java 17.0.6 with PID 1 (/app/identity.jar started by root in /app)                                                                                                                                 
2023-04-21 20:22:10.547  INFO 1 --- [           main] i.c.i.Application                        : The following 1 profile is active: "keycloak"                                                                                                                                                                                 
2023-04-21 20:22:14.753  WARN 1 --- [           main] ocalVariableTableParameterNameDiscoverer : Using deprecated '-debug' fallback for parameter name resolution. Compile the affected code with '-parameters' instead or avoid its introspection:                                                                             io.camunda.identity.config.IdentityCommon                                                                                                                                                                                                                                                                                     
2023-04-21 20:22:14.767  WARN 1 --- [           main] ocalVariableTableParameterNameDiscoverer : Using deprecated '-debug' fallback for parameter name resolution. Compile the affected code with '-parameters' instead or avoid its introspection:                                                                             io.camunda.identity.security.spring.filter.FilterExceptionHandler                                                                                                                                                                                                                                                             
2023-04-21 20:22:15.194  WARN 1 --- [           main] ocalVariableTableParameterNameDiscoverer : Using deprecated '-debug' fallback for parameter name resolution. Compile the affected code with '-parameters' instead or avoid its introspection:                                                                             io.camunda.identity.impl.keycloak.config.record.KeycloakClient                                                                                                                                                                                                                                                                
2023-04-21 20:22:15.604  WARN 1 --- [           main] .s.s.UserDetailsServiceAutoConfiguration :                                                                                                                                                                                                                               

Using generated security password: 9f3cf75e-4ce7-4b9b-a459-871083f4d41b                                                                                                                                                                                                                                                        

This generated password is for development use only. Your security configuration must be updated before running your application in production.                                                                                                                                                                                

2023-04-21 20:22:16.972  INFO 1 --- [           main] i.c.i.Application                        : Started Application in 7.816 seconds (process running for 9.802)                                                                                                                                                              
2023-04-21 20:22:17.922 ERROR 1 --- [           main] i.c.i.i.k.c.KeycloakConfiguration        : Failure #1. Unable to connect to Keycloak.                                                                                                                                                                                    
2023-04-21 20:22:47.930  WARN 1 --- [           main] i.c.i.i.k.c.KeycloakConfiguration        : Retrying...                                                                                                                                                                                                                   
2023-04-21 20:22:47.937 ERROR 1 --- [           main] i.c.i.i.k.c.KeycloakConfiguration        : Failure #2. Unable to connect to Keycloak.                                                                                                                                                                                    
2023-04-21 20:23:17.937  WARN 1 --- [           main] i.c.i.i.k.c.KeycloakConfiguration        : Retrying...                                                                                                                                                                                                                   
2023-04-21 20:23:17.943 ERROR 1 --- [           main] i.c.i.i.k.c.KeycloakConfiguration        : Failure #3. Unable to connect to Keycloak.                                                                                                                                                                                    
2023-04-21 20:23:47.944  WARN 1 --- [           main] i.c.i.i.k.c.KeycloakConfiguration        : Retrying...                                                                                                                                                                                                                   
2023-04-21 20:23:47.949 ERROR 1 --- [           main] i.c.i.i.k.c.KeycloakConfiguration        : Failure #4. Unable to connect to Keycloak.                                                                                                                                                                                    
2023-04-21 20:24:17.950  WARN 1 --- [           main] i.c.i.i.k.c.KeycloakConfiguration        : Retrying...                                                                                                                                                                                                                   
2023-04-21 20:24:17.956 ERROR 1 --- [           main] i.c.i.i.k.c.KeycloakConfiguration        : Failure #5. Unable to connect to Keycloak.                                                                                                                                                                                    
2023-04-21 20:24:47.956  WARN 1 --- [           main] i.c.i.i.k.c.KeycloakConfiguration        : Retrying...                                                                                                                                                                                                                   
2023-04-21 20:24:47.961 ERROR 1 --- [           main] i.c.i.i.k.c.KeycloakConfiguration        : Failure #6. Unable to connect to Keycloak.                                                                                                                                                                                    
2023-04-21 20:24:47.962 ERROR 1 --- [           main] i.c.i.i.k.c.KeycloakConfiguration        : javax.ws.rs.NotFoundException: HTTP 404 Not Found
```

After that I checked the environment variables inside the identity pod while trying to get ready but the KEYCLOAK related variables look OK to me:
```
KEYCLOAK_PORT_80_TCP_PORT=80
KEYCLOAK_PORT_80_TCP_PROTO=tcp
KEYCLOAK_USERS_0_EMAIL=demo@example.org
KEYCLOAK_CLIENTS_0_ROOT_URL=http://placeholder
KEYCLOAK_INIT_TASKLIST_ROOT_URL=http://localhost:8082
KEYCLOAK_INIT_WEBMODELER_ROOT_URL=http://localhost:8084
KEYCLOAK_INIT_OPTIMIZE_ROOT_URL=http://localhost:8083
KEYCLOAK_SETUP_PASSWORD=***MASKED_BUT_OK***
KEYCLOAK_PORT_80_TCP=tcp://10.0.53.41:80
KEYCLOAK_SETUP_USER=admin
KEYCLOAK_CLIENTS_0_PERMISSIONS_0_DEFINITION=read:*
KEYCLOAK_INIT_OPERATE_SECRET=camunda-platform-secrets
KEYCLOAK_SERVICE_PORT_HTTP=80
KEYCLOAK_USERS_0_USERNAME=demo
KEYCLOAK_USERS_0_ROLES_0=Identity
KEYCLOAK_USERS_0_ROLES_1=Operate
KEYCLOAK_CLIENTS_0_NAME=Connectors
KEYCLOAK_USERS_0_ROLES_2=Tasklist
KEYCLOAK_USERS_0_ROLES_3=Optimize
KEYCLOAK_USERS_0_ROLES_4=Web Modeler
KEYCLOAK_USERS_0_LAST_NAME=User
KEYCLOAK_SERVICE_HOST=10.0.53.41
KEYCLOAK_CLIENTS_0_PERMISSIONS_0_RESOURCE_SERVER_ID=operate-api
KEYCLOAK_URL=http://keycloak:80/auth
KEYCLOAK_INIT_OPERATE_ROOT_URL=http://localhost:8081
KEYCLOAK_USERS_0_FIRST_NAME=Demo
KEYCLOAK_CLIENTS_0_ID=connectors
KEYCLOAK_CLIENTS_0_REDIRECT_URIS_0=/
KEYCLOAK_CLIENTS_0_SECRET=camunda-platform-secrets
KEYCLOAK_SERVICE_PORT=80
KEYCLOAK_PORT=tcp://10.0.53.41:80
KEYCLOAK_INIT_TASKLIST_SECRET=camunda-platform-secrets
KEYCLOAK_INIT_OPTIMIZE_SECRET=7BQ7sKcm70
KEYCLOAK_USERS_0_PASSWORD=demo
KEYCLOAK_PORT_80_TCP_ADDR=10.0.53.41
```

My major problem is, that I couldn't find any example, neither a way to enable a higher loglevel for identity to see more about the current error. I did a curl check from an additional pod in the cluster via curl and there I got the standard landing page of keycloak.

```
$> curl keycloak:80
[...]
    <title>Welcome to Keycloak</title>
[...]
```

**Expected behavior:**

Identity is creating a new realm within the keycloak with the given admin user (environment variable `KEYCLOAK_SETUP_USER`) and identity will get up and running and ready to use.

**Environment:**

- Platform: Azure
- Kubernetes: 1.26
- Helm CLI version: "v3.9.0
- Chart version: 8.2.0
- Values file: see relevant informations above
- Keycloak Version: 21

**Comment by knottnhupfer** [comment]
Updated the configuration as following:
```
global:
  identity:
    keycloak:
      auth:
        adminUser: "admin"
        existingSecret: camunda-platform-secrets
        existingSecretKey: "keycloak-secret"
      contextPath: "/"
```

and changed keycloak to version 19, looks everything is working now but the connectors container is still trying to connect to `camunda-platform-key`:
```
Caused by: java.net.UnknownHostException: camunda-platform-key
```
how can I change this URL? Looks like the global configuration is not taken.

**Comment by aabouzaid** [comment]
Hi @knottnhupfer :wave: 

Thanks for reporting this.
Could you please share the final/active values file you have to reproduce the issue?

**Comment by hostalp** [comment]
The problem with the used service and host name being `camunda-platform-key` (instead of `camunda-platform-keycloak`) comes from here:
https://github.com/camunda/camunda-platform-helm/blob/camunda-platform-8.2.0/charts/camunda-platform/templates/_helpers.tpl#L108
Try setting `global.identity.keycloak.legacy` to false if you use Keycloak v19.
Otherwise you may try to use a shorter release name such as `camunda-plm` or just `camunda` which should fit into the 20 characters limit with the "-keycloak" suffix added.

**Comment by aabouzaid** [comment]
In that case, probably it's related to #624
I'm on the fix now.

**Comment by aabouzaid** [comment]
The fix is done, and it will be released in v8.2.1.
